jupyterlab/handlers/ydoc_handler.py

```diff
@@ -12,33 +13,48 @@
 from tornado import web
 from tornado.websocket import WebSocketHandler
 from ypy_websocket.websocket_server import WebsocketServer, YRoom
+from ypy_websocket.ystore import BaseYStore, SQLiteYStore, TempFileYStore, YDocNotFound
 
 YFILE = YDOCS["file"]
-
 RENAME_SESSION = 127
 
 
+class JupyterTempFileYStore(TempFileYStore):
+    prefix_dir = "jupyter_ystore_"
```

Comment:
Why do you use class variables? You do not control their life cycle. It will be better to pass a dictionary of configurable kwargs to the constructor.

Reply:
The problem is that a new Y store is instantiated for each document, identifying the document with its [path](https://github.com/y-crdt/ypy-websocket/blob/8d898b70f05d1c5f6e74521d3f48b497f5bc73ca/ypy_websocket/ystore.py#L18). But you want to store the document updates to some "common root": the database for an [SQLiteYStore](https://github.com/y-crdt/ypy-websocket/blob/8d898b70f05d1c5f6e74521d3f48b497f5bc73ca/ypy_websocket/ystore.py#L100), or a directory prefix for a [TempFileYStore](https://github.com/y-crdt/ypy-websocket/blob/8d898b70f05d1c5f6e74521d3f48b497f5bc73ca/ypy_websocket/ystore.py#L74). So you cannot pass this common root in the constructor, hence the class variable.
There is not issue of class variable life cycle if you create a new class for your store, which is what the docstrings suggest [here](https://github.com/y-crdt/ypy-websocket/blob/8d898b70f05d1c5f6e74521d3f48b497f5bc73ca/ypy_websocket/ystore.py#L67-L72) and [there](https://github.com/y-crdt/ypy-websocket/blob/8d898b70f05d1c5f6e74521d3f48b497f5bc73ca/ypy_websocket/ystore.py#L93-L98).